pkg/controller/controller.go

```diff
@@ -185,6 +192,14 @@ func (c *controller) Run(workers int, stopCh <-chan struct{}) {
 		}
 	}
 
+	func() {
```

Comment:
This is still in the Run method with no context and no comment explaining why it is here. can a worker have a lifecycle that is shorter than the parent process? It feels like this should just be another worker.

Reply:
extracted and commented.